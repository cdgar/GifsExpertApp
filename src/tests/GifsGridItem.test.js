
import React from 'react';
import {shallow} from 'enzyme';
import GifGridItem from '../components/GifGridItem';

describe('GifsGridItem.tets.js', () => {
    let wrapper = shallow(<GifGridItem/>);

    const title = 'titleXD';
    const url = 'urlXD';

    beforeEach(() => {
        wrapper = shallow(<GifGridItem url={url} title={title}/>);
    });

    test('should show GifGridItem', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should get text(title) from paragraph', () => {
        const p = wrapper.find('p').text();
        expect(p).toBe(title);
    });

    test('should img get URL', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should className card animate__animated animate__headShake', () => {
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__headShake')).toBe(true)
    })
    
});