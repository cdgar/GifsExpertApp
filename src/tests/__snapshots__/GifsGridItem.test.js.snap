// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifsGridItem.tets.js should show GifGridItem 1`] = `
<div
  className="card animate__animated animate__headShake"
>
  <img
    alt="titleXD"
    height="100"
    src="urlXD"
    width="200"
  />
  <p>
    titleXD
  </p>
</div>
`;
